src/Query.php

```diff
@@ -495,7 +495,24 @@ public function exists($db = null)
         if (!empty($this->emulateExecution)) {
             return false;
         }
+
+        #better performance
+        #save last options
+        $tmpOrderBy = $this->orderBy;
+        $tmpLimit = $this->limit;
+        $tmpOffset = $this->offset;
+        $tmpSelect = $this->select;
+        $this->orderBy = [];
+        $this->limit = 1;
+        $this->offset = null;
+        $this->select = ['_id'];
         $cursor = $this->buildCursor($db);
+        #return last options
```

Comment:
Why do we need to do it?

Reply:
because objects in `php`  are passed by references by default.
```php
$user = $userRef = User::Find()->Select(['id','name','family']);
$userRef->Where(['=','id',1]);
$user->Select(['id','name']);
$user->Exists();
$userRef->One(); //return last options is useful here
```

if we not want use `tmp` variables then structure of `buildCursor()` must be change.

Reply:
How about using `clone`?